### PR TITLE
Add support of @RunOnVirtualThread on class for websockets next

### PR DESCRIPTION
**Pull request by Malandril**
This PR allows to configure the virtual thread execution model of the websockets next endpoints on the class with `@RunOnVirtualThread` .

```java
    @RunOnVirtualThread
    @WebSocket(path = "/chat")
    public class ChatWebsocket {
        @OnTextMessage
        String text(String message) {
          ....
        }
    }
```

### CODE REVIEW

**Review by mkouba — COMMENTED** [review]
Looks good but we'll need to update the docs as well: https://github.com/quarkusio/quarkus/blob/main/docs/src/main/asciidoc/websockets-next-reference.adoc?plain=1#L233-L251

**Comment by Malandril** [comment]
@mkouba I was wondering also, if it should fail if the annoted Class contains non-blocking methods ?

File: docs/src/main/asciidoc/websockets-next-reference.adoc [hunk 1]
```diff
@@ -240,6 +240,7 @@ WebSocket Next supports _blocking_ and _non-blocking_ logic, akin to Quarkus RES
 Here are the rules governing execution:
 
 * Methods annotated with `@RunOnVirtualThread`, `@Blocking` or `@Transactional` are considered blocking.
+* All Methods in a class annotated with `@RunOnVirtualThread` are considered blocking and run on a virtual thread.
```
Comment (mkouba) [inline docs/src/main/asciidoc/websockets-next-reference.adoc:243]:
```suggestion
* Methods declated on a class annotated with `@RunOnVirtualThread` are considered blocking.
```

Comment (mkouba) [reply docs/src/main/asciidoc/websockets-next-reference.adoc:243]: We should also update the bullet about Kotlin `suspend` functions...

Plus change the last bullet from _"Methods annotated with `@RunOnVirtualThread` must execute on a virtual thread, each invocation spawns a new virtual thread."_ to something like _"Methods annotated with `@RunOnVirtualThread` or declared on a class annotated with `@RunOnVirtualThread` must be executed on a virtual thread, each invocation spawns a new virtual thread."_

**Comment by mkouba** [comment]
> @mkouba I was wondering also, if it should fail if the annoted Class contains non-blocking methods ?

Hm, we don't have this validation at the moment. I mean you can annotate a method that returns `Uni` and we don't fail :thinking:. But we can add this validation later.

**Review by mkouba — APPROVED** [review]

**Comment by mkouba** [comment]
@Malandril merged, thanks for your contribution!

**Pull request merged**